### PR TITLE
Make the task definition optional for the DO_WHILE task

### PR DESCRIPTION
**Pull request by rickfish**
As a System task, the DO_WHILE task should not be required to have a task definition as it makes it difficult to have it not time out. This PR makes it optional.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java [hunk 1]
```diff
@@ -127,7 +129,13 @@ boolean markLoopTaskSuccess(Task task) {
 
 	@VisibleForTesting
 	boolean getEvaluatedCondition(Workflow workflow, Task task, WorkflowExecutor workflowExecutor) throws ScriptException {
-		TaskDef taskDefinition = workflowExecutor.getTaskDefinition(task);
+		TaskDef taskDefinition = null;
```
[the remark this answers is not in the document]
Comment (rickfish) [reply core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java:132]: @manan164 this seems to work fine. the ParameterUtils.getTaskInputV2() method checks for the existence of taskDefinition and works fine if there isn't one. My change does not stop you from having a TaskDef; it just makes it optional. Maybe I am missing something. I did try with a loop condition that uses a variable from task input and the task has no task def and it works as expected. I also added a test case with no task def and it passes.

**Comment by apanicker-nflx** [comment]
@manan164 @rickfish Are we good to merge here?

**Comment by rickfish** [comment]
I believe so

**Comment by manan164** [comment]
Yeah @apanicker-nflx , we are good to merge.
Thanks @rickfish

**Pull request merged**